src/components/TextField/tests/TextField.test.tsx

```diff
@@ -576,13 +643,17 @@ describe('<TextField />', () => {
             autoComplete="off"
           />,
         );
-        element.find('[role="button"]').first().simulate('click');
+        element!
+          .find('div', {
+            role: 'button',
```

Comment:
I find this a bit jarring. Why would a div have a role of button on it? Looking at the `TextField` component it looks like the role doesn't get passed to the div?

Reply:
I was not 100% of the mechanics of it getting passed down. This solution was from converting 
```js
element.find('[role="button"]').first().simulate('click');
```

from the legacy suite and then debugging to find the specific element that the above (and similar) are targeting. 

The debug output for these (`console.log(element.find('[role="button"]').first().debug());`) returns the div element with this role:

```html
<div role="button" ...
```

Looking at it closer, it looks like this is rendered by the `Spinner` component, which accepts a `handleNumberChange` callback, which itself accepts a number (1 for increment, -1 for decrement) and then calls the `onChange` which is being tested.

I believe the `role` is to indicate the div is being used a button. 

One option would be to refactor these test selectors to directly call `onChange` on the <Spinner> rendered in the TestField, as reaching down into the sub-component feels like behaviour that is/should be tested within the Spinner itself.


e.g.
```js
element.find(Spinner)!.trigger('onChange', 1);
```

This proposed change would most likely require additional changes in `Spinner.test.tsx` tests, which currently tests the callback props are called, but does not test if they are called with any of the hardcoded values defined on the Spinner div buttons(1 or -1).